src/app/components/FilterPopup/FilterPopup.jsx

```diff
@@ -141,8 +152,68 @@ class FilterPopup extends React.Component {
     this.setState({ selectedFilters });
   }
 
+  /*
+   * getRaisedErrors(errors)
+   * There's a set list of inputs in the filter form. If the key errors from the form
+   * are found in the set list, it will render those errors. This is meant to be an
+   * aggregate list that is displayed at the top of the form.
+   * @param {Array} errors - An array of the objects with key/value pair of input elements in the
+   *   filter form that have incorrect input.
+   * @return {Array}
+   */
+  getRaisedErrors(errors) {
+    const headlineError = {
+      date: 'dateAfter',
+    };
+    const raisedErrors = [];
+
+    if (!errors || _isEmpty(errors)) {
+      return null;
+    }
+
+    _map(errors, (val, key) => {
+      raisedErrors.push(<li key={key}><a href={`#${headlineError[val.name]}`}>{val.value}</a></li>);
+    });
+
+    return raisedErrors;
+  }
+
+  /*
+   * validateFilterValue(filterValue)
+   * Checks if the values from the input fields are valid. If no, updates the state.
+   *
+   * @param {Object} filterValue
+   * @return {Boolean}
+   */
+  validateFilterValue(filterValue) {
+    const filterErrors = [];
+
+    // If the date input values are invalid
+    if (Number(filterValue.dateBefore) < Number(filterValue.dateAfter)) {
+      const dateInputError = {
+        name: 'date',
+        value: 'Date',
+      };
+
+      filterErrors.push(dateInputError);
+    }
+
+    this.setState({ raisedErrors: filterErrors });
+
+    if (!_isEmpty(filterErrors)) {
```

Comment:
`filterErrors` is an Array and `_isEmpty` works for this, but also check for `.length`.